assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+            if (!$scope.buildFrom.imageStreams.contains($scope.options.pickedBuildFromImageStream)) {
+              $scope.options.pickedBuildFromImageStreamTag = "";
+            }
+            if (selectFirstOption) {
+              $scope.options.pickedBuildFromImageStream = $scope.buildFrom.imageStreams[0];
+              $scope.clearSelectedBuilderTag();
+            }
+          } else {
+            $scope.options.pickedBuildFromImageStream = "";
+            $scope.options.pickedBuildFromImageStreamTag = "";
+          }
+        });
+      }
+    }
+
+    $scope.clearSelectedBuilderTag = function() {
+      var tags = $scope.buildFrom.tags[$scope.options.pickedBuildFromImageStream];
+      $scope.options.pickedBuildFromImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0] || "";
+    }
+
+    // updateOutputImageStreams creates/updates the list of imageStreams and imageStreamTags for the output image from picked namespace.
+    // As parameter takes the picked namespace, selectFirstOption as a boolean that indicates whether the imageStreams and imageStreamTags
+    // selectboxes should be select the first option.
+    $scope.updateOutputImageStreams = function(projectName, selectFirstOption) {
+      DataService.list("imagestreams", {namespace: projectName}, function(imageStreams) {
+        $scope.pushTo.imageStreams = [];
+        $scope.pushTo.tags = {};
+        var projectImageStreams = imageStreams.by("metadata.name");
+        if (!_.isEmpty(projectImageStreams)) {
+          angular.forEach(projectImageStreams, function(imageStream, name) {
+            var tagList = [];
+            if (imageStream.status.tags) {
+              imageStream.status.tags.forEach(function(item){
+                tagList.push(item["tag"]);
+              });
+            }
+            $scope.pushTo.imageStreams.push(name);
+            $scope.pushTo.tags[name] = tagList;
+          });
+
+          if (selectFirstOption) {
+            $scope.options.pickedPushToImageStream = $scope.pushTo.imageStreams[0];
+            $scope.clearSelectedOutputTag();
+            // If defined Image Stream is not present in the Image Streams of picked Namespace set tag to empty string,
+            // so the user has to pick from a existing Image Streams.
+          } else if (!$scope.pushTo.imageStreams.contains($scope.options.pickedPushToImageStream)) {
+            $scope.options.pickedPushToImageStreamTag = "";
+          }
+        } else {
+          $scope.options.pickedPushToImageStream = "";
+          $scope.options.pickedPushToImageStreamTag = "";
+        }
+      });
+    }
+
+    $scope.clearSelectedOutputTag = function() {
+      var tags = $scope.pushTo.tags[$scope.options.pickedPushToImageStream];
+      $scope.options.pickedPushToImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0] || "latest";
+    }
+
+    $scope.save = function() {
```

Comment:
I tend to like to see save functions for controllers delegate to services as well.  Can still return the `DataService.update()` promise to give control to UI:

```javascript
$scope.save = function() {
 // BuildConfigService update will return the underlying DataService.update('buildconfigs', buildConfigName, ...)
 BuildConfigService
    .update(buildConfigName, $scope.updatedBuildConfig) // hide the details of processing this from the controller
   .then(function() {
     // add success alert
   }, function() {
     // add error alert
   });
}
```